stock_picking_invoicing/models/stock_move.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2019-Today: Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class StockMove(models.Model):
+    _name = "stock.move"
+    _inherit = [
+        _name,
+        "stock.invoice.state.mixin",
+    ]
+
+    def _get_price_unit_invoice(self, inv_type, partner, qty=1):
+        """
+        Gets price unit for invoice
+        :param inv_type: str
+        :param partner: res.partner
+        :param qty: float
+        :return: float
+        """
+        product = self.mapped("product_id")
+        product.ensure_one()
+        if inv_type in ("in_invoice", "in_refund"):
+            result = product.standard_price
+        else:
+            # If partner given, search price in its sale pricelist
+            if partner and partner.property_product_pricelist:
+                product = product.with_context(
+                    partner=partner.id,
+                    quantity=qty,
+                    pricelist=partner.property_product_pricelist.id,
+                    uom=fields.first(self).product_uom.id,
+                )
+                result = product.lst_price
+            else:
+                result = product.lst_price
```

Comment:
This always returns product.lst_price.
Getting price from pricelist is rather complicated https://github.com/odoo/odoo/blob/d44106309c4a11fc3c0dca4ed8457e1f27bd1527/addons/sale/models/sale_order_line.py#L427
My idea of re implementing pricelist support in this module seems bad. I suggest using account_invoice_pricelist if the project requires it.

Keeping this method as hook for other modules is a good idea though.

```suggestion
        else:
                result = product.lst_price
```